### PR TITLE
[FEAT]: more sql functions

### PR DESCRIPTION
**Pull request by universalmind303**
fills out the remaining numerics, and adds most of the utf8 functions. 

Also adds a few helper methods to `Expr` and `LiteralValue` for extracting the underlying scalar values.

### CODE REVIEW

File: src/daft-sql/src/functions.rs [hunk 1]
```diff
@@ -12,27 +13,357 @@ use crate::{
...
+    // ------------------------------------------------
+    // Aggregate Functions
+    // ------------------------------------------------
     Max,
 }
 
 impl FromStr for SQLFunctions {
     type Err = PlannerError;
 
     fn from_str(value: &str) -> Result<Self, Self::Err> {
```
Comment (kevinzwang) [inline src/daft-sql/src/functions.rs:309]: If you are tired of writing all the match arms by hand we could use [strum](https://docs.rs/strum_macros/latest/strum_macros/derive.EnumString.html) to auto-generate this!

File: src/daft-dsl/src/lit.rs [hunk 2]
```diff
@@ -246,6 +246,71 @@ impl LiteralValue {
             Python(..) => display_sql_err,
         }
     }
+
+    /// If the liter is a boolean, return it. Otherwise, return None.
+    pub fn as_bool(&self) -> Option<bool> {
```
Comment (kevinzwang) [inline src/daft-dsl/src/lit.rs:251]: Would it be worth writing a macro for all these `as_` functions? Seems like a good amount of boilerplate

File: src/daft-sql/src/functions.rs [hunk 3]
```diff
@@ -78,38 +409,276 @@ impl SQLPlanner {
...
+            }
+            SQLFunctions::Ln => {
+                ensure!(args.len() == 1, "ln takes exactly one argument");
+                Ok(daft_dsl::functions::numeric::ln(args[0].clone()))
+            }
+            SQLFunctions::Log => {
+                ensure!(args.len() == 2, "log takes exactly two arguments");
+                let base = args[1]
+                    .as_literal()
+                    .and_then(|lit| lit.as_f64())
```
Comment (kevinzwang) [inline src/daft-sql/src/functions.rs:503]: log base could be an integer value too right?

**Review by kevinzwang — COMMENTED** [review]
Some small comments

File: src/daft-dsl/src/lit.rs (hunk 2 above)
Comment (universalmind303) [reply src/daft-dsl/src/lit.rs:252]: some of them such as primitives (bool, i32, ...) dereference, but others return a reference, so I think the extra boilerplate is likely needed.

**Review by kevinzwang — APPROVED** [review]
LGTM!

**Pull request merged**